src/libraries/System.Private.CoreLib/src/System/Collections/BitArray.cs

```diff
@@ -943,6 +943,48 @@ public bool HasAnySet()
             return (_array[byteCount] & mask) != 0;
         }
 
+        /// <summary>Computes the number of bits that are set in the <see cref="BitArray"/>.</summary>
+        /// <returns>The number of set bits in the <see cref="BitArray"/>.</returns>
+        public int PopCount()
+        {
+            int count = 0;
+            foreach (byte b in _array)
+                count += byte.PopCount(b);
+            return count;
+        }
+
+        /// <summary>Computes the number of trailing zero bits in the <see cref="BitArray"/>.</summary>
+        /// <returns>The number of trailing zero bits in the <see cref="BitArray"/>.</returns>
+        public int TrailingZeroCount()
+        {
+            int count = 0;
+            for (int i = 0; i < _array.Length; i++)
+            {
+                if (_array[i] != 0)
```

Comment:
This is iterating from beginning to end. Shouldn't it be going from end to beginning? Or have I misunderstood?